### ISSUE TITLE
Is it possible to supplement the training files

### ISSUE THREAD

**Issue by XieJiajian2024** [goglxych97/iterU-Net#1]
Thank you for your great work! May I ask if you can supplement the relevant code for model training and testing, especially for this two-stage training process? Thanks again!

**Comment by goglxych97** [comment]
Thank you very much for your interest in this work.
I’ve uploaded a Jupyter Notebook that covers the overall process for both warm-up training and optimization training.
For testing, you can simply use the same validation procedure as in the optimization stage.
Please feel free to let me know if you find anything unclear or if there’s anything that could be improved.

Thanks again for your kind interest, and I hope you have a great day!

**Comment by XieJiajian2024** [comment]
@goglxych97   Thank you very, very much!
And, is iterUNet used for the first stage warm up model? I noticed that the 75 line code x_post=iter_outputs [-1] used the previous output of the model. I hope I have not misunderstood!
Thanks again!😊

**Comment by goglxych97** [comment]
Thank you very much for your thoughtful question, and I really appreciate your careful review of the code.
You are absolutely right, there was a mistake in the code. The warm-up model is trained on the real images before being trained to the fine-tuning stage (optimization). The overall idea is to first pretrain the model on the real images and then refine it using the outputs generated during the optimization process. I’ve updated this part in the code to reflect this correction.

Thank you again for catching that, and I sincerely apologize for the confusion.

**Comment by XieJiajian2024** [comment]
@goglxych97 Thank you! 
May I ask on what memory GPU are you training the model? Even though I reduced the number of channels by half, it still exceeded 24GB of memory

**Comment by wJJin** [comment]
Thank you for your kind interest in this work!
My training environment is as follows:

Training Environment
- GPU: NVIDIA GeForce RTX 4090 (24GB)
- Driver Version: 535.183.06
- CUDA Version: 12.2
- PyTorch: 2.1.2+cu121
- MONAI: 1.2.dev2312

Model Configuration
- Channels: (28, 32, 64, 128, 256, 28)
- Input Patch Size: 96 × 96 × 48 (original: 560 × 560 × 160)
- Time Steps: 6 in total (pre-contrast, post-1, post-2, post-3, post-4, post-5)

Training Setup
- Warm-up phase: Batch size = 2 (negative + positive patch per sample, total 4 patches)
- Optimize phase: Batch size = 1 (negative + positive patch per sample, total 2 patches)
- Loss functions: combination of L1, SSIM, perceptual, and gradient difference losses

GPU Memory Usage: As shown in the figure below.

<img width="1393" height="766" alt="Image" src="https://github.com/user-attachments/assets/a11f9b8a-bf5d-4edc-a5b9-2608720170bc" />

<br>
<br>
It might be related to differences in GPU environments or how memory is managed during training. <br>
I hope this helps and that it works well on your setup.

**Comment by XieJiajian2024** [comment]
@wJJin @goglxych97  Thank you for your detailed answers all along! 
Perhaps it's because my dataset is much larger. Again, thank you very much for your open source spirit.This is very meaningful
